### PR TITLE
Let coral hive2rel treat a union typeinfo as a struct typeinfo, inste…

### PR DESCRIPTION
**Pull request by rzhang10**
…ad of throwing exception

Also added `extract_union` function to static registry.

Tested on hive view which contains `extract_union` and coral can translate it correctly to spark sql.

### CODE REVIEW

**Comment by wmoustafa** [comment]
Could you please add unit tests?

**Comment by shenodaguirguis** [comment]
+1 to @wmoustafa's request to add tests. LGTM o.w.

**Review by shenodaguirguis — APPROVED** [review]
thanks for adding the unit tests.

**Comment by funcheetah** [comment]
@rzhang10 @shenodaguirguis  is this PR integration tested for all views since it is not a trivial change?

File: coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java [hunk 1]
```diff
@@ -316,6 +316,7 @@ public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
 
     createAddUserDefinedFunction("array_contains", ReturnTypes.BOOLEAN, family(SqlTypeFamily.ARRAY, SqlTypeFamily.ANY));
     createAddUserDefinedFunction("sort_array", ARG0, ARRAY);
+    createAddUserDefinedFunction("extract_union", ARG0, or(ANY, family(SqlTypeFamily.ANY, SqlTypeFamily.INTEGER)));
```
Comment (wmoustafa) [inline coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java:319]: For the 2-argument case, the output type will not be `ARG0`. Also, for the 1-argument case, the type would always be a struct, but it seems in the test, we define the table as containing `uniontype` in the tests, and hence the output could be a `uniontype` if we use `ARG0`?

**Pull request merged**